backend/infrahub/core/manager.py

```diff
@@ -218,6 +261,8 @@ async def count(
 
         if isinstance(schema, str):
             schema = db.schema.get(name=schema, branch=branch.name)
+        elif hasattr(schema, "_is_runtime_protocol") and getattr(schema, "_is_runtime_protocol"):
+            schema = db.schema.get(name=schema.__name__, branch=branch.name)
```

Comment:
Doesn't have to be within this PR but here we could perhaps also have a function to get the schema that's shared between .count() and .query().

Reply:
Good point